### PR TITLE
VideoPlayer: register ProcessInfo

### PR DESCRIPTION
**Pull request by FernetMenta**

### CODE REVIEW

File: xbmc/cores/VideoPlayer/Process/ProcessInfo.h [hunk 1]
```diff
@@ -25,15 +25,20 @@
 #include "threads/CriticalSection.h"
 #include <atomic>
 #include <list>
+#include <map>
 #include <string>
 
+class CProcessInfo;
 class CDataCacheCore;
 
+typedef CProcessInfo* (*CreateProcessControl)();
```
Comment (garbear) [inline xbmc/cores/VideoPlayer/Process/ProcessInfo.h]: we've switch to the c++ `using` syntax, which I think works for function types too

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/Process/ProcessInfo.h]: I was never en expert in regard to function pointers. how would it look like with "using"?

Comment (garbear) [inline xbmc/cores/VideoPlayer/Process/ProcessInfo.h]: `using CreateProcessControl = CProcessInfo* (*)();` should be correct. I ran a quick test and switching to this syntax compiles successfully.

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/Process/ProcessInfo.h]: that looks much nicer. I will try. Thanks a lot

**Pull request merged**